xBrokerConnector/src/main/java/dev/midas/xbc/stream/DisposableFactory.java

```diff
@@ -0,0 +1,106 @@
+package dev.midas.xbc.stream;
+
+import com.google.common.collect.FluentIterable;
+import info.bitrich.xchangestream.core.StreamingExchange;
+import info.bitrich.xchangestream.core.StreamingExchangeFactory;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.disposables.Disposables;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ *  This factory currently creates and destroys markets streams.
+ */
+@Component
+public class DisposableFactory {
```

Comment:
`DisposableFactory` is more than just a factory and needs a better name 😛

Reply:
I totally agree.